run_bot_streaming.sh

```
@@ -0,0 +1,209 @@
+    local processed_config=$(process_config "$config_json")
+
+    # Get recording and debug settings
+    local recording_mode=${RECORDING:-true}
+    local debug_mode=${DEBUG:-false}
+    local debug_logs=${DEBUG_LOGS:-false}
+
+    print_info "Recording enabled: $recording_mode"
+    print_info "Output directory: $output_dir"
+
+    # Find available port for bot API
+    local main_port=$(find_available_port 3000)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)